app/controllers/application_controller.rb

```diff
@@ -8,4 +10,11 @@ def after_sign_in_path_for(resource)
   def after_sign_out_path_for(resource)
     root_path
   end
+
+  private
+    def user_not_authorized
+      flash[:alert] = "You are not authorized to perform this action."
+
+      redirect_back fallback_location: root_url
+    end
```

Comment:
Nice job implementing Pundit to prevent unauthorized users from accessing routes.